panos_upgrade_assurance/snapshot_compare.py

```diff
@@ -129,30 +129,38 @@ def compare_snapshots(
         return result
 
     @staticmethod
-    def key_checker(left_dict: dict, right_dict: dict, key: str) -> None:
-        """The static method to check if a key is available in both dictionaries.
+    def key_checker(left_dict: dict, right_dict: dict, key: Union[str, set, list]) -> None:
+        """The static method to check if a key(string) or list/set of keys is available in both dictionaries.
 
-        This method looks for a given key in two dictionaries. Its main purpose is to assure that when comparing a key-value pair from two dictionaries, it actually exists in both.
+        This method looks for a given key or list/set of keys in two dictionaries. Its main purpose is to assure that when comparing a key-value pair from two dictionaries, it actually exists in both.
 
         # Parameters
 
         left_dict (dict): 1st dictionary to verify.
         right_dict (dict): 2nd dictionary to verify.
-        key (str): Key name to check.
+        key (str, set, list): Key name or set/list of keys to check.
 
         # Raises
 
         MissingKeyException: when key is not available in at least one snapshot.
 
         """
-        left_snap_missing_key = False if key in left_dict else True
-        right_snap_missing_key = False if key in right_dict else True
+
+        if isinstance(key, str):
+            key_set = [key]
+        elif isinstance(key, (set, list)):
+            key_set = key
```

Comment:
Actually this is not a set, it's a list. You do convert it to a set later on, so maybe let's do it here? Then we can simplify the `if` statement to something like:

```suggestion
        if isinstance(key, (str, set, list)):
            key_set = set(key)
```

Reply:
The thing with str is if you cast it to a `set` directly it splits every character of the string as separate elements of the set. e.g. for `key="tcp"` it becomes `{'t','c','p'}`. We need to cast str to a list first and then to set like `set([key])`. However we can move the casting to set to here instead of the if conditions below, I will update that.